src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -17,7 +23,31 @@ public LineService(LineRepository lineRepository) {
     }
 
     public LineResponse saveLine(LineRequest request) {
+        if (lineRepository.findByName(request.getName()).isPresent()) {
+            throw new DuplicateLineException("중복된 지하철 노선 이름이 있습니다.");
+        }
         Line persistLine = lineRepository.save(request.toLine());
         return LineResponse.of(persistLine);
     }
+
+    public List<LineResponse> findAllLines() {
+        return lineRepository.findAll()
+          .stream()
+          .map(LineResponse::of)
+          .collect(Collectors.toList());
+    }
+
+    public LineResponse findById(Long id) {
+        Line line = lineRepository.findById(id).orElseThrow(() -> new NoSuchElementException("노선을 찾을 수 없습니다."));
+        return LineResponse.of(line);
```

Comment:
하나의 라인에 하나의 점을 쓰면 가독성이 올라갑니다 :)

추가적으로 NoSuchElementException 보다는 EntityNotFoundException를 사용해보는건 어떨까요?
```suggestion
        Line line = lineRepository.findById(id)
                .orElseThrow(() -> new NoSuchElementException("노선을 찾을 수 없습니다."));
```

Reply:
앗 가독성을 높이도록 해야겠네요 ㅎㅎ `EntityNotFoundException`이 있군요..! 좋은 말씀 감사합니다!